uint8[] pinNumber
---
bool[] pinValue